easybuild/easyblocks/l/llvmcore.py

```diff
@@ -0,0 +1,748 @@
+#   /home/ampandey/rocm-toolchain/build/…/external/llvm-project/llvm
+
+
+# https://github.com/llvm/llvm-project/issues/72108
+# -DCMAKE_BUILD_TYPE="Release" \
+# -DCMAKE_C_FLAGS="-Wno-backend-plugin" \
+# -DCMAKE_CXX_FLAGS="-Wno-backend-plugin" \
+# -DCMAKE_C_COMPILER="clang" \
+# -DCMAKE_CXX_COMPILER="clang++" \
+# -DLLVM_ENABLE_CLASSIC_FLANG=ON \
+# -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;lld;openmp" \
+# -DLLVM_ENABLE_RUNTIMES="compiler-rt;libcxx;libcxxabi" \
+# -DLLVM_LIBDIR_SUFFIX=64 \
+# -DLLVM_USE_LINKER=lld \
+# -DLLVM_PARALLEL_COMPILE_JOBS="96" \
+# -DLLVM_PARALLEL_LINK_JOBS=96 \
+# -DLLVM_ENABLE_NEW_PASS_MANAGER=ON \
+# -DLLVM_TARGETS_TO_BUILD="X86;BPF;WebAssembly" \
+# -DLLVM_ENABLE_LIBCXX=OFF \
+# -DLLVM_STATIC_LINK_CXX_STDLIB=OFF \
+# -DLLVM_BINUTILS_INCDIR="/usr/include" \
+# -DLLVM_ENABLE_TERMINFO=OFF \
+# -DLLVM_ENABLE_LIBXML2=OFF \
+# -DLLVM_ENABLE_LIBEDIT=OFF \
+# -DLLVM_ENABLE_PER_TARGET_RUNTIME_DIR=OFF \
+# -DCOMPILER_RT_BUILD_LIBFUZZER=OFF \
+# -DCOMPILER_RT_BUILD_XRAY=OFF \
+# -DCOMPILER_RT_BUILD_ORC=OFF \
+# -DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON \
+# -DLIBCXX_ENABLE_STATIC_ABI_LIBRARY=OFF \
+# -DOPENMP_ENABLE_LIBOMPTARGET=OFF \
+# -DOPENMP_ENABLE_OMPT_TOOLS=OFF \
+# -DOPENMP_ENABLE_TESTING=OFF \
+# -DCOMPILER_RT_CXX_LIBRARY=libcxx \
+# -DSANITIZER_CXX_ABI=libc++ \
+# -DSANITIZER_CXX_ABI_INTREE=ON \
+# -DSANITIZER_USE_STATIC_CXX_ABI=ON \
+# -DSANITIZER_TEST_CXX=libc++ \
+# -DSANITIZER_TEST_CXX_INTREE=ON \
+# -DSANITIZER_USE_STATIC_TEST_CXX=ON \
+# -DSANITIZER_LIT_USE_LIBCXX=ON \
+
+remove_gcc_opts = {
+    # 'LLVM_LIBC_FULL_BUILD': 'On',
+    # 'LLVM_LIBC_INCLUDE_SCUDO': 'On',
+
+    'LIBCXX_USE_COMPILER_RT': 'On',
+    'LIBCXX_CXX_ABI': 'libcxxabi',
+
+    'LIBCXXABI_USE_LLVM_UNWINDER': 'On',
+    'LIBCXXABI_USE_COMPILER_RT': 'On',
+
+    'LIBUNWIND_USE_COMPILER_RT': 'On',
+
+    'SANITIZER_USE_STATIC_LLVM_UNWINDER': 'On',
+    # 'SANITIZER_USE_STATIC_CXX_ABI': 'On',
+    'COMPILER_RT_USE_LIBCXX': 'On',
+    'COMPILER_RT_USE_LLVM_UNWINDER': 'On',
+    # 'COMPILER_RT_CXX_LIBRARY': 'libcxx',
+    'COMPILER_RT_USE_BUILTINS_LIBRARY': 'On',
+    'COMPILER_RT_ENABLE_STATIC_UNWINDER': 'On',  # https://lists.llvm.org/pipermail/llvm-bugs/2016-July/048424.html
+    'COMPILER_RT_ENABLE_INTERNAL_SYMBOLIZER': 'On',
+    'COMPILER_RT_BUILD_GWP_ASAN': 'Off',
+    # 'COMPILER_RT_BUILD_SCUDO_STANDALONE_WITH_LLVM_LIBC': 'On',
+    # 'COMPILER_RT_SCUDO_STANDALONE_BUILD_SHARED': 'Off',
+    # Required for building the standalone libatomic (not depending on GCCcore)
+    # 'COMPILER_RT_BUILD_STANDALONE_LIBATOMIC': 'On',
+
+    'CLANG_DEFAULT_CXX_STDLIB': 'libc++',
+    'CLANG_DEFAULT_RTLIB': 'compiler-rt',
+    'CLANG_DEFAULT_LINKER': 'lld',
+    'CLANG_DEFAULT_UNWINDLIB': 'libunwind',
+
+    'LIBCXX_HAS_GCC_S_LIB': 'Off',
+    'LIBCXXABI_HAS_GCC_S_LIB': 'Off',
+    'LIBUNWIND_HAS_GCC_S_LIB': 'Off',
+    # 'COMPILER_RT_HAS_GCC_S_LIB': 'Off',
+    # 'CLANG_HAS_GCC_S_LIB': 'Off',
+
+    # 'OPENMP_ENABLE_OMPT_TOOLS': 'Off',
```

Comment:
For the EasyBlock, I would give users the choice to disable the OpenMP Tools Interface, if they wish to do so. There is little reason to do so, and I would leave it on by default, but maybe users are interested in disabling it.
That option should exist even when not doing a full independent LLVM build though.

Reply:
Yes, I agree. I've done the change in https://github.com/easybuilders/easybuild-easyblocks/pull/3373/commits/e18fedd093bfc34cec85130ce58a65aa90f8b666